audit_manager.tf

```diff
@@ -0,0 +1,41 @@
+resource "aws_auditmanager_account_registration" "default" {
+  count = var.aws_auditmanager_config.enabled == true ? 1 : 0
+
+  delegated_admin_account = data.aws_caller_identity.audit.account_id
+  deregister_on_destroy   = true
+  kms_key                 = module.kms_key_audit.arn
+
+  depends_on = [
+    module.kms_key_audit
+  ]
+}
+
+module "audit_manager_reports" {
+  count     = var.aws_auditmanager_config.enabled == true ? 1 : 0
+  providers = { aws = aws.audit }
+
+  source      = "schubergphilis/mcaf-s3/aws"
+  version     = "0.12.1"
+  name_prefix = var.aws_auditmanager_config.reports_bucket_name
+  versioning  = true
+
+  lifecycle_rule = [
+    {
+      id      = "retention"
+      enabled = true
+
+      abort_incomplete_multipart_upload = {
+        days_after_initiation = 7
+      }
+
+      noncurrent_version_expiration = {
+        noncurrent_days = 90
```

Comment:
does expiration after 90 days make sense for audit manager reports?

Reply:
This is for non-current (e.g deleted and outdated) versions of the objects. I think it's alright, can you do a suggestion if you have another idea?

Reply:
If we don't need them we can also drop the whole lifecycle_rule property

Reply:
Ah I misread the lifecycle rule, sgtm